zanata-war/src/main/java/org/zanata/dao/ApplicationConfigurationDAO.java

```diff
@@ -1,12 +1,14 @@
 package org.zanata.dao;
 
 import org.hibernate.Session;
+
+import javax.enterprise.context.RequestScoped;
 import javax.inject.Named;
 import org.zanata.model.HApplicationConfiguration;
 
 @Named("applicationConfigurationDAO")
 
-@javax.enterprise.context.Dependent
+@RequestScoped
```

Comment:
Not sure about this change alex. Maybe I'm being overly cautious, but DAOs I think should remain dependent scoped as they should not have any state, hence any instance of them should work regardless of the context. Could you elaborate on how this fixes the problem?